src/QMCTools/XmlRep.cpp

```diff
@@ -147,9 +147,9 @@ int XmlStream::addNextTag()
 {
   std::streampos start;
   std::streampos end;
-  char twoprev = '0';
-  char prev = '0';
-  char current = '0';
+  int twoprev = '0';
```

Comment:
Please explain this change. Character-like usage appears below. Also, =0 not ='0' if truly an int.

Reply:
It should be null char 0 not '0'. Clang complains about comparing char with EOF (-1).
current, prev, twoprev are return values of istream::get() which is int not char. I think this is intended by the standard to communicate more info rather than just a character.
Comparing integer with normal char is allowed by the language.